website/blog/2024-03-11-AutoDefense/Defending LLMs Against Jailbreak Attacks with AutoDefense.mdx

```diff
@@ -0,0 +1,91 @@
+
+1. **Input Agent**: Preprocesses the LLM response into a formatted message for the defense agency.
+2. **Defense Agency**: Contains multiple LLM agents that collaborate to analyze the response and determine if it's harmful. Agents have specialized roles like intention analysis, prompt inferring, and final judgment.
+3. **Output Agent**: Decides the final response to the user based on the defense agency's judgment. If deemed harmful, it overrides with an explicit refusal.
+
+The number of agents in the defense agency is flexible. We explore configurations with 1-3 agents. 
+
+![image-20240306194957182](imgs/image-20240306194957182.png)
+
+### Defense Agency
+
+The three-agent defense agency consists of:
+
+1. **Intention Analyzer**: Analyzes the intention behind the given content. This helps identify potentially malicious motives.
+2. **Prompt Analyzer**: Infers possible original prompts that lead to the response.  By reconstructing these prompts without jailbreak elements, it activates the LLMs' safety mechanisms.
+3. **Judge**: Makes the final judgment on whether the response is harmful based on the analysis from the intention and prompt analyzers.
+
+A coordinator agent manages the workflow between these agents, passing messages and prompting each to perform their specialized task. This division of roles makes it easier for LLMs with limited steerability to follow instructions for their specific sub-task, resulting in a more accurate overall judgment.
+
+## Experiment Setup
+
+We evaluate AutoDefense on two datasets:
+
+- Curated set of 33 harmful prompts and 33 safe prompts. Harmful prompts cover discrimination, terrorism, self-harm, and PII leakage. Safe prompts are GPT-4 generated daily life and science inquiries.
+- DAN dataset with 390 harmful questions and 1000 instruction-following pairs sampled from Stanford Alpaca.
+
+Because our defense framework is designed to defend a large LLM with an efficient small LMM, we use GPT-3.5 as the victim LLM in our experiment.
+
+We use different types and sizes of LLMs to power agents in the multi-agent defense system:
+
+1. **GPT-3.5-Turbo-1106**
+2. **LLaMA-2**: LLaMA-2-7b, LLaMA-2-13b, LLaMA-2-70b
+3. **Vicuna**: Vicuna-v1.5-7b, Vicuna-v1.5-13b, Vicuna-v1.3-33b
+4. **Mixtral**: Mixtral-8x7b-v0.1, Mistral-7b-v0.2
+
+We use llama-cpp-python to serve the chat completion API for open-source LLMs, allowing each LLM agent to perform inference through a unified API. INT8 quantization is used for efficiency.
+
+LLM temperature is set to `0.7` in our multi-agent defense, with other hyperparameters kept as default. 
+
+## Experiment Results
+
+![image-20240311175815518](imgs/image-20240311175815518.png)
+
+We compare different methods for defending GPT-3.5 as shown in Table 3. The LLaMA-2-13B is used as the defense LLM in AutoDefense. We find our AutoDefense outperforms other methods in terms of ASR.
```

Comment:
Add a overall sentence to introduce the two experiments we have: 1. Compare number of agents. 2. Compare with other defenses.